packages/core/src/config/dragConfigResolver.ts

```diff
@@ -26,8 +27,18 @@ export const dragConfigResolver = {
     if (SUPPORT.touch) return 'touch'
     return 'mouse'
   },
-  preventScroll(value = false) {
-    return value && SUPPORT.touch
+  preventScroll(value: number | boolean = 0, _k: string, { preventScrollAxis = 'x' }) {
+    if (preventScrollAxis) this.preventScrollAxis = preventScrollAxis
```

Comment:
TypeScript should complain here as `this` is not defined. In the function arguments, pass `this` as in:

```ts
preventScroll(this: InternalDragOptions, value: number | boolean = 0, _k: string, { preventScrollAxis = 'x' })
//            ^^^^
```
You'll also need to type the config passed as the third argument as `preventScrollAxis` will be typed as a string and not `x` | `y`. Bottom line:

```ts
  preventScroll(
    this: InternalDragOptions,
    value: number | boolean = 0,
    _k: string,
    { preventScrollAxis = 'x' }: DragConfig
//                               ^^^^^^^^^^
  )
```